File: app/src/main/java/com/uiic/dwellingpremium/MainActivity.java
package com.uiic.dwellingpremium;

import android.content.Intent;
import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.Spinner;
import android.widget.Toast;

import androidx.appcompat.app.AppCompatActivity;

import com.google.android.material.textfield.TextInputLayout;
import com.uiic.dwellingpremium.model.Premium;
import com.uiic.dwellingpremium.utils.DwellingConstants;

public class MainActivity extends AppCompatActivity {

    private static final String LOG_TAG = MainActivity.class.getSimpleName();

    TextInputLayout sumTextInputLayout;
    TextInputLayout yearsTextInputLayout;
    EditText sumInsuredEditText;
    EditText yearsInsuredEditText;
    Spinner schemeSpinner;
    Button button;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);


        sumTextInputLayout = findViewById(R.id.edit_layout_sum_insured);
        yearsTextInputLayout = findViewById(R.id.edit_layout_years_insured);
        sumInsuredEditText = findViewById(R.id.edit_sum_insured);
        yearsInsuredEditText = findViewById(R.id.edit_years_insured);
        schemeSpinner = findViewById(R.id.spinner_premium_scheme);
        button = findViewById(R.id.main_cal_premium_button);

        button.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                if (sumInsuredEditText.getText().toString().isEmpty()) {
                    sumTextInputLayout.setError("Sum Insured cannot be empty");
                } else if (Long.parseLong(sumInsuredEditText.getText().toString()) == 0) {
                    sumTextInputLayout.setError("Invalid Value");
                } else if (yearsInsuredEditText.getText().toString().isEmpty()) {
                    yearsTextInputLayout.setError("Years cannot be empty");
                } else if (Integer.parseInt(yearsInsuredEditText.getText().toString()) == 0 || Integer.parseInt(yearsInsuredEditText.getText().toString()) < 1) {
                    yearsTextInputLayout.setError("Invalid Value");
                } else {
                    try {
                        Log.i(LOG_TAG, "SumInsured and Years have been entered");
                        double sumInsured = Double.parseDouble(sumInsuredEditText.getText().toString());
                        int yearsInsured = Integer.parseInt(yearsInsuredEditText.getText().toString());
                        String selectedScheme = schemeSpinner.getSelectedItem().toString();

                        Premium premium = new Premium(sumInsured, yearsInsured, selectedScheme);

                        Intent premiumIntent = new Intent(view.getContext(), PremiumActivity.class);
                        premiumIntent.putExtra(DwellingConstants.PREMIUM_OBJECT_KEY, premium);
                        premiumIntent.putExtra(DwellingConstants.YEARS_INSURED_KEY, yearsInsured);
                        premiumIntent.putExtra(DwellingConstants.SCHEME_KEY, selectedScheme);

                        startActivity(premiumIntent);

                    } catch (Exception e) {
                        Log.e(LOG_TAG, "Error occurred in parsing input string");
                        e.printStackTrace();
                        Toast.makeText(MainActivity.this, "Error : Check Input String", Toast.LENGTH_LONG).show();
                    }
                }
            }
        });
    }
}

File: app/src/main/java/com/cyorg/uiic/dwellingpremium/activity/MainActivity.kt
package com.cyorg.uiic.dwellingpremium.activity

import android.content.Context
import android.content.Intent
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.util.Log
import com.cyorg.uiic.dwellingpremium.*
import com.cyorg.uiic.dwellingpremium.model.Premium
import com.cyorg.uiic.dwellingpremium.utils.DwellingConstants
import kotlinx.android.synthetic.main.activity_main.*

/**
 * Created by yorga on 07-07-2017.
 */
class MainActivity : AppCompatActivity() {

    private val LOG_TAG: String = MainActivity::class.java.simpleName
    private var activityContext: Context? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        activityContext = this
        setContentView(R.layout.activity_main)

        main_cal_premium_button.setOnClickListener {

            var validSumInsured: Boolean = edit_sum_insured.isValidLong()
            var validYearsInsured: Boolean = edit_years_insured.isValidInt()
            var selectedScheme = spinner_premium_scheme.selectedItem as String


            if (!validSumInsured)
                edit_layout_sum_insured.error = "Invalid Value"
            if (!validYearsInsured)
                edit_layout_years_insured.error = "Invalid Value"

            if (validSumInsured && validYearsInsured) {
                try {
                    Log.i(LOG_TAG, "SumInsured and Years have been entered")
                    val sumInsured = edit_sum_insured.toLong()
                    val yearsInsured = edit_years_insured.toInt()

                    val premiumModel = Premium(sumInsured, yearsInsured)
                    premiumModel.calculatePremium(selectedScheme)

                    val premiumIntent = Intent(activityContext, PremiumActivity::class.java)
                    premiumIntent.putExtra(DwellingConstants.PREMIUM_OBJECT_KEY, premiumModel)
                    premiumIntent.putExtra(DwellingConstants.YEARS_INSURED_KEY, yearsInsured)
                    premiumIntent.putExtra(DwellingConstants.SCHEME_KEY, selectedScheme)

                    startActivity(premiumIntent)

                } catch (e: Exception) {
                    Log.e(LOG_TAG, "Error occurred in parsing input string")
                    e.printStackTrace()
                }

            }

        }
    }
}